chalice/pipeline.py

```diff
@@ -9,18 +9,36 @@
 def create_pipeline_template(config):
     # type: (Config) -> Dict[str, Any]
     pipeline = CreatePipelineTemplate()
-    return pipeline.create_template(config.app_name)
+    return pipeline.create_template(config.app_name,
+                                    config.lambda_python_version)
+
+
+class InvalidCodeBuildPythonVersion(Exception):
+    def __init__(self, version):
+        # type: (str) -> None
+        super(InvalidCodeBuildPythonVersion, self).__init__(
+            'CodeBuild does not yet support python version %s.' % version
+        )
 
 
 class CreatePipelineTemplate(object):
 
+    _CODEBUILD_IMAGE = {
+        'python2.7': 'python:2.7.12',
+    }
+
     _BASE_TEMPLATE = {
         "AWSTemplateFormatVersion": "2010-09-09",
         "Parameters": {
             "ApplicationName": {
                 "Default": "ChaliceApp",
                 "Type": "String",
                 "Description": "Enter the name of your application"
+            },
+            "CodeBuildImage": {
```

Comment:
Yea thats a good point. I could pass through some cli argument like `--codebuild-image` which would override whatever the auto selected one. Which would be nice in case a user has their own python 3.6 image. Although the only use case I can think of for this is to use your own image when CodeBuild hasn't caught up to Lambda, so the reason for this flag's existence would have to be documented well.